scipy/optimize/_cobyqa_py.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+
+from ._optimize import _check_unknown_options
+
+__all__ = []
+
+
+def _minimize_cobyqa(fun, x0, args=(), bounds=None, constraints=(),
+                     callback=None, disp=False, maxfev=1000, maxiter=1000,
+                     target=-np.inf, feasibility_tol=1e-8,
```

Comment:
Could you add a test for the `target` termination criterion? That one should be easy to do. I guess the other tolerances are hard to test without reusing COBYQA internals. I think we should rely on tests in the actual COBYQA repo for that, do you agree?

Reply:
Yes, I agree. I did not include all the tests in the actual COBYQA repo to SciPy because they take around 20 minutes to finish. For `feasibility_tol`, we can test that `maxcv` in the result is below this value if `success` is `True`.

Reply:
This should be done in e737b9c1518aed996a4798b94ce1bc67692cc780. Concerning the tests for `feasibility_tol`, there were already present because I test that `sol.maxcv < 1e-8`, which is the default `feasibility_tol`. I let you resolve the conversation is you are happy with the modifications.

Reply:
Thanks for the new test, looks good!

I had edited my first message a few minutes after it was first posted and added the first paragraph about the naming of the parameter. Your update was already on the way, so you probably missed that part (my bad actually, should have probably added it below the original comment, not above ... ). Could you still adapt that?

Reply:
Indeed, I did not see the update. I had already created a test for `target`, now renamed `f_min`. Does it look ok to you?

Reply:
Hi @dschmitz89 and @ragonneau,

> What is called target here is called f_min for [SHGO](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.shgo.html#scipy.optimize.shgo) and [DIRECT](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.direct.html#scipy.optimize.direct). For consistency let us stick with the same parameter name here.

Indeed, I would suggest the opposite. As a person working in optimization for more than 15 years, I have never heard anyone meant target value by f_min, which exclusively mean the minimum of f. To be frank, this is a wrong name in the current context. 

In contrast, ftarget or f_target is the standard term adopted in popular optimization packages. @ragonneau please put some references here.

Thanks.

Zaikun

Reply:
Hi @zaikunzhang,

I suppose you are referring to the following.
1. BFO (https://sites.google.com/site/bfocode) from Ph. L. Toint and M. Porcelli.
2. A. Friedlander and J. M. Martínez. Convergence and complexity in nonlinear optimization. Course dispensed at Unicamp, São Paulo, Brazil. 2017. url: https://www.ic.fcen.uba.ar/elavio/friedlandermartinez.pdf.
3. R. Tanabe. "Benchmarking feature-based algorithm selection systems for black-box numerical optimization." IEEE Trans. Evol. Comput. 26.6 (2022), pp. 1321–1335. doi: [10.1109/TEVC.2022.3169770](https://doi.org/10.1109/TEVC.2022.3169770).

Note that they referred to it as `ftarget` and not `f_target`. It is clear that `ftarget` here is not possible as it does not really comply with the Python naming standard [PEP8](https://peps.python.org/pep-0008/).

Cheers,
Tom.